### ISSUE TITLE
Documentation for capacity-limit-backup-calculator and capacity-limit values/functions

### ISSUE THREAD

**Issue by robbieorvis** [JGCRI/gcam-doc#111]
I'm trying to better understand how the default values for backup capacity were determined for GCAM and how users can change these. I know that that the capacity limit can be changed directly via the XML files and that insert a value can modify the curvature, but I'm hoping to find the equations for this function so I can better modify the function. The docs say it is documented in the electricity section but it's not really discussed and there are no equations.

**Comment by ssmithClimate** [comment]
You can see the formula used here:
https://github.com/JGCRI/gcam-core/blob/master/cvs/objects/sectors/source/capacity_limit_backup_calculator.cpp
(line 220)

While this is named capacity limit, this can be thought of as a cost adder that represents generalized costs of adding variable renewable capacity. There are four parameters that are set as defaults, but users can read-in other values to change the shape of the curve.

**Comment by robbieorvis** [comment]
Thanks @ssmithClimate. I assume those 4 parameters could be read-in as part of capacity-limit-backup-calculator parameters in the appropriate XML files? E.g.:

<capacity-limit-backup-calaculator>
   <CapacityLimit></CapacityLimit>
   <Fmax></Fmax>
   <C></C>
   <Tau></Tau>
</capacity-limit-backup-calculator>

Or is there another format that is needed?

Haewon sent me something at one point that noted I could change the <C/> parameter.

So you are aware, the current default parameters can cause extreme discontinuities and failure to solve from simply clearing a certain threshold of clean energy. Once that threshold is breached the electricity costs skyrocket and demand plummets. This does not seem like rational behavior in the power sector, hence my desire to play around with some of these parameters. Changing the <C/> value has helped, but I haven't really understood the mechanism.

Since this is an important aspect of the power sector selection and generation, it would be really great if you could document in the GCAM docs why this exists and how it works.

**Comment by robbieorvis** [comment]
@ssmithClimate Would you be able to take a look at the attached and let me know if I'm doing the calculation correctly? I believe the y-axis is GW backup to GW added capacity[
[Capacity Backup Calculator Function.xlsx](https://github.com/JGCRI/gcam-doc/files/4828830/Capacity.Backup.Calculator.Function.xlsx)
](url)

**Comment by amnmalik** [comment]
Hi @robbieorvis! Could you better understand the backup capacity/costs in the meantime? I am facing the same issue of abrupt changes in backup as RE penetration increases.

**Comment by robbieorvis** [comment]
Yes, @d3y419 helped me figure it out. I'm dropping our email thread below for you. I've bolded the parts with the actual solution.

Good to know.

I think this one can stay off github for now. I think we’d actually need to properly model the curtailment (as opposed to just backup).

Best,

Haewon.

From: Robbie Orvis <[robbie@energyinnovation.org](mailto:robbie@energyinnovation.org)> 
Sent: Friday, December 13, 2019 1:38 PM
To: McJeon, Haewon C <[Haewon.McJeon@pnnl.gov](mailto:Haewon.McJeon@pnnl.gov)>
Subject: RE: Question on changing backup capacity limits

Just circling back this – this was a perfect fix.

I set **<c>0.5</c>** and the problem vanished. I still see the point at which backup becomes a requirement but it is a smoothed transition.

FWIW, I also think this is much more representative of how the power system actually operates. If you look at curtailment curves/functions, which can sort of be thought of as a proxy for backup requirements, you see that there is some curtailment even at lower penetration levels and this grows sort of linearly through a wide range of the penetration percentage up to a cap. Like an S-curve, but with a pretty shallow shape. There’s not really a penetration percentage where you flip a switch and go from no curtailment/backup requirement to all of a sudden needing one. (Apologies if this is already an issue you are familiar with).

If you think it’s worthwhile, I’m happy to report this as an issue to GitHub, and it looks like it would also avoid weird model behavior in future release.

Thanks again for your help – couldn’t do all this without your guidance!

-R

________________________________________
Robbie Orvis
Director of Energy Policy Design
Phone: 415-799-2171
98 Battery Street, Suite 202
San Francisco, CA  94111
[www.energyinnovation.org](https://protect2.fireeye.com/v1/url?k=fd86ae40-a13391f9-fd868455-0cc47adc5fce-f438dc989707ab7e&q=1&e=58ed0bb5-6bfd-4d99-a57b-d3ed50481cdc&u=http%3A%2F%2Fwww.energyinnovation.org%2F)
 
________________________________________

Check out our new book, [Designing Climate Solutions: A Policy Guide for Low-Carbon Energy](https://www.amazon.com/Designing-Climate-Solutions-Policy-Low-Carbon/dp/1610919564)
Available wherever books are sold

 

From: McJeon, Haewon C <[Haewon.McJeon@pnnl.gov](mailto:Haewon.McJeon@pnnl.gov)> 
Sent: Wednesday, December 11, 2019 11:38 AM
To: Robbie Orvis <[robbie@energyinnovation.org](mailto:robbie@energyinnovation.org)>
Subject: RE: Question on changing backup capacity limits

 

It doesn’t look too bad…

It’s clearly a signature of PV backup kicking in. 

There’s a more detailed query that distinguishes backup vs. capital cost:

<supplyDemandQuery title="costs by tech and input">
    <axis1 name="technology">technology</axis1>
    <axis2 name="Year">price-paid[@vintage]</axis2>
    <xPath buildList="true" dataName="fuel cost" group="false" sumAll="false">*[@type = 'sector']/*[@type = 'subsector']/*[@type = 'technology']/
            *[@type='input']/price-paid[@vintage=parent::*/parent::*/@year]/text()</xPath>
    <comments>excludes energy and emissions related costs which are not printed</comments>
</supplyDemandQuery>

This will show whether indeed the backup is the issue.

The backup curve is indeed very steep. Basically you don’t need any of it, until you need it. And when you need it ramps up pretty quickly. I don’t exactly remember how the curvature parameter works. But you should read in something like this:

                        **<capacity-limit-backup-calculator>
                            <c>1</c>
                        </capacity-limit-backup-calculator>**

Hope this helps!

Haewon.



From: Robbie Orvis <[robbie@energyinnovation.org](mailto:robbie@energyinnovation.org)> 
Sent: Monday, December 9, 2019 5:38 PM
To: McJeon, Haewon C <[Haewon.McJeon@pnnl.gov](mailto:Haewon.McJeon@pnnl.gov)>
Subject: RE: Question on changing backup capacity limits

Hey Haewon,

Thanks for the quick response!

I had thought that could be it, but I was surprised it would have such a strong effect on prices…

Attached is a bunch of output for China, but let me know what other regions and outputs would be most helpful, and I can send that. The backup capacity limit is set to one in this scenario. At higher values, I don’t see this impact (for example, I just ran with a capacity limit of 3, which for solar means 50% of system capacity (30 * 20% capacity factor)… though that seems to have other issues like runaway solar deployment, etc…

I see how to control the limit, but not the curvature. I was wondering about the curvature being too steep and hence causing a dramatic impact once some threshold is hit.

Thanks also for the info and your help on transportation. I haven’t heard anything from Leon since I spoke with him about 3 weeks ago… Hence hiring a consulting firm to help with some of this since we need to get the work done within the next few months. 

Best,

Robbie

________________________________________
Robbie Orvis
Director of Energy Policy Design
Phone: 415-799-2171
98 Battery Street, Suite 202
San Francisco, CA  94111
[www.energyinnovation.org](https://protect2.fireeye.com/v1/url?k=3e32c8fe-6287f747-3e32e2eb-0cc47adc5fce-6f7b1e72e9b3b774&q=1&e=08717405-78b4-4340-a88b-a18dc2f157be&u=http%3A%2F%2Fwww.energyinnovation.org%2F)
 
________________________________________

Check out our new book, [Designing Climate Solutions: A Policy Guide for Low-Carbon Energy](https://www.amazon.com/Designing-Climate-Solutions-Policy-Low-Carbon/dp/1610919564)
Available wherever books are sold

 

From: McJeon, Haewon C <[Haewon.McJeon@pnnl.gov](mailto:Haewon.McJeon@pnnl.gov)> 
Sent: Monday, December 9, 2019 1:54 PM
To: Robbie Orvis <[robbie@energyinnovation.org](mailto:robbie@energyinnovation.org)>
Subject: RE: Question on changing backup capacity limits

Can you send me the data table?

I think that’s the year the “backup” kicks in. you can experiment with the capacity limit and the curvature to control this effect.

All vintages are affected, but only the new vintages are relevant for market share decisions.

Best,

Haewon.

Ps. I’m talking to a few people about the transportation update. Will get to you shortly. The short answer is that we indeed have updated cost and efficiency numbers, we just need to talk to a few people about how and when we can share them.

From: Robbie Orvis <[robbie@energyinnovation.org](mailto:robbie@energyinnovation.org)> 
Sent: Monday, December 9, 2019 3:51 PM
To: McJeon, Haewon C <[Haewon.McJeon@pnnl.gov](mailto:Haewon.McJeon@pnnl.gov)>
Subject: Question on changing backup capacity limits

Hi Haewon,

I’ve been experimenting with changing the backup capacity factors for intermittent techs based on our earlier conversation. I’m seeing this resulting in higher deployment, but it also seems to be causing weird effects on prices. For example, if I raise the backup capacity limit to 1, I finding that in certain years technology prices jump by 10-20% (I’m looking at GCAM global at China). It seems to be affecting all vintages of the given tech. This in turn is increasing the price of electricity, which is then flowing through to the demand sectors, etc…

Do you know why this might be happening or what to do about it?

Thanks.

Best,

**Comment by amnmalik** [comment]
Thanks a lot @robbieorvis !
